File: VB/MyRegistrator/MyBaseTabPageViewInfo.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraTab
Imports DevExpress.XtraTab.ViewInfo
Imports Example_HidePageHeader.MyTabControl
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace Example_HidePageHeader
	Public Class MyBaseTabPageViewInfo
		Inherits BaseTabPageViewInfo
		Public Sub New(ByVal page As IXtraTabPage)
			MyBase.New(page)
		End Sub
		Public Overrides Property Bounds() As System.Drawing.Rectangle
			Get
				Dim nullRectangle = New System.Drawing.Rectangle(MyBase.Bounds.Location, New System.Drawing.Size(0, 0))
				If (CType(Page, MyXtraTabPage)).HidePageHeader Then
					Return nullRectangle
				End If
				Return MyBase.Bounds
			End Get
			Set(ByVal value As System.Drawing.Rectangle)
				MyBase.Bounds = value
			End Set
		End Property
	End Class
End Namespace

File: CS/MyRegistrator/MyBaseTabPageViewInfo.cs
using DevExpress.XtraTab;
using DevExpress.XtraTab.ViewInfo;
using Example_HidePageHeader.MyTabControl;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Example_HidePageHeader {
    public class MyBaseTabPageViewInfo : BaseTabPageViewInfo {
        public MyBaseTabPageViewInfo(IXtraTabPage page)
            : base(page) {
        }
        public override System.Drawing.Rectangle Bounds {
            get {
                var nullRectangle = new System.Drawing.Rectangle(base.Bounds.Location, new System.Drawing.Size(0, 0));
                if(((MyXtraTabPage)Page).HidePageHeader) {
                    return nullRectangle;
                }
                return base.Bounds;
            }
            set {
                base.Bounds = value;
            }
        }
    }
}
